subsys/net/lib/coap/coap_transport_socket.c

```diff
@@ -332,7 +332,7 @@ u32_t coap_transport_write(const coap_transport_handle_t transport,
 					address_length_get(remote));
 		}
 
-		if (retval == -1) {
+		if (retval == -1 && errno != EAGAIN) {
```

Comment:
I think we should forward EAGAIN as a return value to the application. Currently the application couldn't actually tell, whether the data was sent or not in case of non-blocking socket.

Reply:
The current approach seems to be to always return ```EIO``` and let ```errno``` still have the actual errno, which could be read by the application. Do you think we should return ```EAGAIN``` instead of ```EIO``` in this specific case?

Reply:
Well, `coap_transport_write` API does not mention errno at all, it only states that error code is returned. So I think we should return an error code indicating that the data was not sent (EAGAIN in this specific case).

And current approach will return 0 in case `errno == EAGAIN`, so we'd need to verify errno each time in case of success. And as no error condition might have occured, errno can contain rubbish or some old value.